### ISSUE TITLE
[Hold #43508] Workspace - "Hidden" members appear in the list when a member leaves

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#42555]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.75-0
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:** N/A
**Issue reported by:** Applause - Internal Team

**Issue found when executing PR** https://github.com/Expensify/App/pull/41492
## Action Performed:
1. Admin: Log in as a new expensifail user
2. Member: Log in as a new Gmail user in a different device
3. Admin: Create a workspace
4. Admin: Invite the member to the workspace
5. Admin: Navigate to the Workspace member list page
6. Admin: Open the member details page
7. Member: Leave the workspace
8. Admin: Click on "Go back to home page"

## Expected Result:
Only real members of the workspace should be shown

## Actual Result:
"Hidden" members appear in the list when a workspace member leaves

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details> 

https://github.com/Expensify/App/assets/78819774/6638fe81-83c0-4f02-996a-b9fac2ab0b09

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by lanitochka17** [comment]
@trjExpensify  FYI I haven't added the External label as I wasn't 100% sure about this issue. Please take a look and add the label if you agree it's a bug and can be handled by external contributors

**Comment by lanitochka17** [comment]
We think that this bug might be related to #wave-collect - Release 1

**Comment by trjExpensify** [comment]
Commented on the issue [the PR](https://github.com/Expensify/App/pull/41492) is attached to. CC: @alitoshmatov @tienifr @madmax330

**Comment by trjExpensify** [comment]
Alright, so looks like this was missed. Members should be able to leave a workspace unless they're under domain control on both NewDot and OldDot. 

[This issue](https://github.com/Expensify/Expensify/issues/43264#issuecomment-311027460) provides a bit of history on why that wasn't the case with a helpful comment from @iwiznia. That said, a lot has changed since then and Auth is aware of the employeeList now. 👍 

Sending `Internal`.

**Comment by trjExpensify** [comment]
Waiting to be picked up in hot picks.

**Comment by trjExpensify** [comment]
No change.

**Comment by trjExpensify** [comment]
@Beamanator with the approvals work and behaviour around leaving a workspace, will this be cleaned up? I think it's somewhat linked to this weird dynamic of allowing someone to "leave" workspace on NewDot but not properly handled on the backend, as you can't do that on OldDot.

**Comment by Beamanator** [comment]
I thinkkkkk you're right and yes it "should get fixed" with those changes, but I would recommend we keep this open (maybe drop to weekly / monthly) to make sure this doesn't accidentally slip by, unfixed

**Comment by trjExpensify** [comment]
Yep, makes sense. Do you have a particular issue I can point to for the hold?

**Comment by Beamanator** [comment]
possibly probably https://github.com/Expensify/Expensify/issues/404210?

**Comment by trjExpensify** [comment]
Works for me!

**Comment by trjExpensify** [comment]
Okay, the issue linked has merged. Adding `retest weekly` label for QA to re-test this to confirm it has been fixed.

**Comment by trjExpensify** [comment]
Awaiting a retest.

**Comment by trjExpensify** [comment]
Still awaiting a retest

**Comment by mvtglobally** [comment]
Issue is reproducible during KI retests.
![1722864277960!Screenshot_2024-08-05_161845](https://github.com/user-attachments/assets/548dbccd-2898-44da-96eb-c087456d18d3)

**Comment by trjExpensify** [comment]
Okay, so that wasn't fixed then @Beamanator. Let's make sure it's taken care of in [here](https://github.com/Expensify/App/issues/43508) then I guess?

**Comment by trjExpensify** [comment]
Put this on hold for #43508

**Comment by trjExpensify** [comment]
Still held, Melv.

**Comment by trjExpensify** [comment]
No change.

**Comment by trjExpensify** [comment]
We're still held.

**Comment by trjExpensify** [comment]
Still holding.

**Comment by trjExpensify** [comment]
Still held.

**Comment by trjExpensify** [comment]
Still held. I'm dropping to monthly until there's traction on the other.

**Comment by trjExpensify** [comment]
Still held on https://github.com/Expensify/App/issues/43508

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (Second week)

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (Third week)

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (Fourth week)

**Comment by trjExpensify** [comment]
Hasn't been reproducible for 4 weeks, closing! I suspect other clean up of "hidden" resolved this one.
